lib/l10noptimize/parser/index.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var L10nPropParser = require('./l10n-properties');
+var L20nPropParser = require('./l20n-properties');
+
+exports.parse = function (l10nLib, source) {
+  if (l10nLib === 'l10n.js') {
+    return L10nPropParser.parse(source);
+  } else {
+    return L20nPropParser.parse(source);
```

Comment:
Not much. The binary choice between `l10n.js` and `l20n.js` will stay the same. Inside the ParserProxy we will have to recognize if we're in `.l20n` or `.properties` and load appropriate parser, but it we don't need any more bits of information from the HTML file.